doc/_static/pulse/make_pulse_images.py

```diff
@@ -0,0 +1,125 @@
+
+import pennylane as qml
+
+folder = pathlib.Path(__file__).parent
+
+
+def parametrized_coefficients_example(savefile="parametrized_coefficients_example.png"):
+
+    # defining the coefficients fj(v, t) for the two parametrized terms
+    f1 = lambda p, t: p * jnp.sin(t) * (t - 1)
+    f2 = lambda p, t: p[0] * jnp.cos(p[1] * t ** 2)
+
+    # defining the operations for the three terms in the Hamiltonian
+    XX = qml.PauliX(0) @ qml.PauliX(1)
+    YY = qml.PauliY(0) @ qml.PauliY(1)
+    ZZ = qml.PauliZ(0) @ qml.PauliZ(1)
+
+    H1 = 2 * XX + f1 * YY + f2 * ZZ
+
+    times = jnp.linspace(0., 5., 1000)
+    fs = H1.coeffs_parametrized
+    ops = H1.ops_parametrized
+    params = [1.2, [2.3, 3.4]]
+
+    fig, axs = plt.subplots(nrows=len(ops))
+    for n, f in enumerate(fs):
+        ax = axs[n]
+        ax.plot(times, f(params[n], times), label=f"p={params[n]}")
+        ax.set_ylabel(f"{ops[n].label()}")
+        ax.legend(loc="upper left")
+
+    ax.set_xlabel("time")
+    axs[0].set_title(f"H1 parametrized coefficients")
+    plt.tight_layout()
+
+    plt.savefig(folder / savefile)
+    plt.close()
+
+
+def pwc_example(savefile="pwc_example.png"):
+    params = jnp.array([1, 2, 3, 4, 5])
+    time = jnp.linspace(0, 10, 1000)
+    y = qml.pulse.pwc(timespan=(2, 7))(params, time)
+
+    plt.plot(time, y, label=f"params = {params}, \n timespan = (3, 8)")
+    plt.legend(loc="upper left")
+
+    plt.savefig(folder / savefile)
+    plt.close()
+
+
+def rect_example(savefile="rect_example.png"):
+    def f(p, t):
+        return jnp.polyval(p, t)
+
+    p = jnp.array([1, 2, 3])
+    time = jnp.linspace(0, 10, 1000)
+
+    y1 = f(p, time)
+    y2 = [qml.pulse.rect(f, windows=[(1, 7)])(p, t) for t in time]
+
+    plt.plot(time, y1, label=f"polyval, P = {p}")
+    plt.plot(time, y2, label="rect(polyval), windows = [(1, 7)]")
+    plt.legend()
+
+    plt.savefig(folder / savefile)
+    plt.close()
+
+
+def pwc_and_rect_example(savefile="pwc_and_rect.png"):
+    def f(p, t):
+        return jnp.polyval(p, t)
+
+    times = jnp.linspace(0., 10., 1000)
+
+    fig, axs = plt.subplots(1, 2)
+
+    polyval_params = jnp.array([1, 2, 3])
+    windows = [(3, 6)]
+
+    axs[0].plot(times, f(polyval_params, times), label=f"polyval, p = {polyval_params}")
+    axs[0].plot(times, [qml.pulse.rect(f, windows=windows)(polyval_params, t) for t in times],
+                label=f"rect(polyval), \n windows={windows}")
+    axs[0].set_title("rect")
+    axs[0].set_xlabel("time")
+    axs[0].legend()
+
+    axs[1].plot(times, qml.pulse.pwc(windows[0])([1, 2, 3, 4, 5], times),
+                label=f"timespan = {windows[0]}, \n params = [1, 2, 3, 4, 5]")
+    axs[1].set_title("pwc")
+    axs[1].set_xlabel("time")
+    axs[1].legend(loc="upper left")
+
+    plt.tight_layout()
+    plt.savefig(folder / savefile)
+    plt.close()
+
+
+if __name__ == "__main__":
+    parametrized_coefficients_example()
+    pwc_example()
+    rect_example()
+    pwc_and_rect_example()
```

Comment:
Where is this one used? :)

Reply:
It's not, nor will it be 😆  I just thought I might want it at the time.